ble/src/main/java/no/nordicsemi/android/ble/BleManagerCallbacks.java

```diff
@@ -74,8 +74,38 @@
 	 * Otherwise the {@link #onLinkLossOccurred(BluetoothDevice)} method will be called instead.
 	 *
 	 * @param device the device that got disconnected.
+	 * @deprecated Use {@link #onDeviceDisconnected(BluetoothDevice, int)} instead.
 	 */
-	void onDeviceDisconnected(@NonNull final BluetoothDevice device);
+	@Deprecated
+	default void onDeviceDisconnected(@NonNull final BluetoothDevice device) {
+		// do nothing
+	}
+
+	/**
+	 * Called when the device has disconnected (when the callback returned
+	 * {@link BluetoothGattCallback#onConnectionStateChange(BluetoothGatt, int status, int)} with
+	 * newstate DISCONNECTED), but ONLY if the {@link ConnectRequest#shouldAutoConnect()} method
+	 * returned false for this device when it was connecting.
+	 * Otherwise the {@link #onLinkLossOccurred(BluetoothDevice)} method will be called instead.
```

Comment:
I think here you want to reference the onLinkLossOccurred method with int parameter

Reply:
True, but...